### DISCUSSION TITLE
Will Urql cache the request if response is single null field?

### DISCUSSION THREAD

**Discussion by Trav1sRen** [urql-graphql/urql#2148] [Q&A] [answered]
I have a query to check the login state like below:
`
Query LoginState() {
  loginState() {
    id,
    username
  }
}
`

And the logic in resolver is:
if browser session has no wanted property --> return null,
if browser session has wanted property --> return the user object.

The flow on the FrontEnd is:
1. Clear the session and make the call to LoginState query, which returns
`
data {
  loginState: null
}
`
2. Sequentially make another same call

My question is, will the Urql client cache the first call which returns null (which means the second call will return null as well)?
Or the first call won't be cached and Urql will make another call for it.

Now what I observed is: the first call which returns null is not cached.

Im new to the GraphQL, could anyone help to explain the inner logic of the Urql implementation of this part?

**Comment by JoviDeCroock** [comment, marked as answer]
It depends is the best answer, are you returning an `Error` alongside this?

In urql we have two caches the default (document-cache) and the normalized cache (graphcache).

In doc-cache any query with `data` gets cached according to their `body-variables` unique key https://github.com/FormidableLabs/urql/blob/main/packages/core/src/exchanges/cache.ts#L140-L149

In GraphCache we'll leverage your `GraphQL` errors to see whether the path returned in `errors` are null, this means that we can partially cache said query given the error-paths and when the query isn't complete we'll redispatch a request.

**Reply by Trav1sRen** [reply]
Im using 'type-graphql' and 'apollo-server-express' for my BackEnd.
Here is the simple resolver code:
![image](https://user-images.githubusercontent.com/63448147/145786264-4a4d821e-ee82-46e7-a2c6-213159ccd01d.png)

And the User type is defined here:
![image](https://user-images.githubusercontent.com/63448147/145786365-f55f1297-aa5b-42ee-a370-b784cbb6157e.png)

And I did not customize the `exchanges` in `createClient` options.

So in that case, the Urql SHOULD cache the said null query if im using the default requestPolicy 'cache-first'.
Am I correct?
